src/mds/Server.cc

```diff
@@ -1658,6 +1658,16 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
   }
 }
 
+bool Server::is_operation_on_dot_or_dotdot(const filepath& path) {
+  std::string dname = path.last_dentry();
+  if (dname == "." || dname == "..") {
+    dout(20) << __func__ << ": operation is on . or .." << dendl;
+    return true;
+  }
+
+  return false;
+}
```

Comment:
I think we can add is_last_dot_or_dotdot() helper to src/include/filepath.h

Reply:
I kept it here as it was the only place that needed the checks.

Reply:
need to check path.depth()  >0